woocommerce/woocommerce
# Address some deprecation warnings

### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:

This PR address 3 different deprecation warnings:
- The use of the conditional block
- Using a very old version of React context ( which wasn't being used anymore anyhow )
- Remove use of position in `Dropdown` usage.



### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

**NOTE:** there are still a couple other warnings present, one is for the `tooltip` usage, but that only got deprecated in 6.4, so we may have to wait with fixing that until `6.5` is out.

**Testing link update**
1. Load this branch and enable the new product editor in **WooCommerce > Settings > Advanced > Features**
2. Open the browser console
3. Go to **Products > Add new**
4. Click the **Pricing** link in the list price
5. It should navigate to the price tab correctly without errors
6. Check your console, there should not be a warning present saying: `Legacy context API has been detected within a strict-mode tree`

**Testing conditional block**
1. Load this branch and enable the new product editor in **WooCommerce > Settings > Advanced > Features**
2. Open the browser console
3. Go to **Products > Add new**
4. Click the **Inventory** tab
5. **Track stock...** should be disabled
6. The **Stock status** and under advanced the **Restrictions** field should be present
7. Enable **Track stock...**
8. The **Available quantity** field should show and the **Stock status** should dissapear.
9. Under advanced the **When out of stock** and **email me when....** should should be present.
10. It should navigate to the price tab correctly without errors
11. Check your console, there should not be a warning present saying: `'woocommerce/conditional' block is deprecated. Please use 'hideConditions' attribute on any block instead.`

**Testing dropdown usage**
1. Load this branch and enable the new product editor in **WooCommerce > Settings > Advanced > Features**
2. Open the browser console
3. Go to **Products > Add new**
4. Click the Downloads dropdown and use the **Insert from URL** ( should work correctly )
6. Create a product with variations and use the dropdown filters in the variation table, they should work correctly.
8. Check your console, there should not be a warning present saying: `'position' prop in wp.components.Dropdown is deprecated since version 6.2. Please use 'popoverProps.placement' prop instead. Note: Note that the 'position' prop will override any values passed through the 'popoverProps.placement' prop.`
12. Also try the dropdowns usage in Analytics > Products filtering by date or **Analytics > Settings** using the default date range dropdown.
13. Check if the above warning from step 6 isn't shown.




### Changelog entry




<details>

#### Significance




#### Type




#### Message 

#### Comment 

</details>
